services/basic_types.proto

```diff
@@ -1592,3 +1593,32 @@ message StakingInfo {
         int64 staked_node_id = 6;
     }
 }
+
+message VirtualAddress {
+    /**
+     * The 20-byte EVM address.
+     */
+    bytes address = 1;
+
+    /**
+     * The ethereum transaction nonce associated with this address.
+     */
+    int64 nonce = 2;
+
+    /**
+     * Flag if this address should now be set or is the default address on the account.
+     */
+    bool is_default_address = 3;
+}
+
+message VirtualAddressUpdate {
+    /**
+     * An accounts 20-byte EVM address account identifier and nonce.
+     */
+    VirtualAddress virtual_address = 1;
```

Comment:
So to remove a VirtualAddress you need to provide a nonce and possibly a default flag? Bit weird even if those values might be ignored for removals. Another possible simplification is to have the field names make the action explicit:

```proto
VirtualAddress virtual_address_add = 1;
bytes virtual_address_remove = 2;
```

or

```proto
oneof virtual_address {
    VirtualAddress add = 1;
    bytes remove = 2;
}

Reply:
No on deletion the nonce would be null.
The goal was to share the message object and populate the appropriate items on input or output

Reply:
Also the VirtualAddress was also intended to expose the newest nonce value to the mirror node on transactions. In which case it's neither an addition nor removal of a virtual address. Just an update to the evmAddresses nonce.

Reply:
because nonce is an `int64` there is no null, an absent field becomes `0`, which may be sufficient for our needs.  If we need to distinguish from null and 0 we need `Int64Value`

Reply:
Yeah `Int64Value` would have been better.
However, seems like no nonce is the way and the `oneof virtual_address_update` suffices.

Reply:
Updated.